### PR TITLE
Enable integrations

### PR DESCRIPTION
**Pull request by hx-markterry**
#### What does this PR do? (please provide any background)

We would like to have more than just the linting checks behind makeup, dependency checking and CSS checks should be arriving soon.

In order to support more checks this work refactors how the current lint checking is called.

I'll be bumping versions in a separate PR.

#### What tests does this PR have?

Unit tests for changed functionality as well as some more tests to increase the overall coverage of this project.

#### How can this be tested?

Running the following will ensure the code still works as before:

    npm run lint

#### What gif best describes how you feel about this work?

giphy

---

- [x] I have checked the [contributing document](../blob/master/CONTRIBUTING.md) and I'm happy for this to be reviewed.

#### Reviewers

**Review 1**
- [x] :+1:

**Review 2** \*
- [x] :+1:

**Review 3** _(optional)_
- [ ] :+1:

By adding a +1 you are confirming you have...
- Witnessed the work behaving as expected (this could be on the author's machine or screencast).
- Checked for coding anti-patterns.
- Checked for appropriate test coverage.
- Checked all the tests are passing.

\*  for HX this review must be completed by an SE, SA or Project Guru

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,9 +56,16 @@ Alternatively, to only check files newer than a specific git history date, use t
...
+## Extending
+
+To add a new integration to this project follow the steps below:
 
-Make sure nothings broken before pushing `$ npm test` is good at that.
+1. Create a new js module in the `lib/integrations` directory.
+1. Make sure this module exports an object.
+1. Create a function named `run` on the new integration object, this will take two arguments; an `options` object and `callback` function.
+1. The `run` function will run the given callback in the usual node manner with the first argument being set as an error object if needed.
+1. The `run` function will output any detail via `console.log` statements.
```
Comment (pmcnr-hx) [inline README.md]: Would it make sense to instead pass a stream to log to (which would default to `process.stderr` as a configuration property)?

**Comment by pmcnr-hx** [comment]
I like this! :+1:

**Comment by GeorgeF72** [comment]
1st

**Comment by GeorgeF72** [comment]
Looks good to me :+1:

**Comment by pmcnr-hx** [comment]
Sorry the late comment, but got a lot of distractions since our chat @hx-markterry. The change to using streams looks marvellous: :+1: :+1: :shipit:

**Pull request merged**